### PR TITLE
6.2 Feature. Change behavior of Sprint Active/Inactive sprints

### PR DESCRIPTION
**Pull request by sub13**

### CODE REVIEW

File: src/Teams.Web/Controllers/ManageSprintsController.cs [hunk 1]
```diff
@@ -142,34 +132,44 @@ public async Task<IActionResult> EditSprintAsync(int teamId, int sprintId, strin
...
-        public async Task<IActionResult> AddSprintAsync(int teamId, string sprintName, int daysInSprint, int storePointsInHours, bool isActive)
+        public async Task<IActionResult> AddSprintAsync(int teamId, string sprintName, int daysInSprint, int storePointsInHours, int status)
         {
-            var sprint = new Sprint { TeamId = teamId, Name = sprintName, DaysInSprint = daysInSprint, StoryPointInHours = storePointsInHours, IsActive = isActive };
-            
+            var sprint = new Sprint { TeamId = teamId, Name = sprintName, DaysInSprint = daysInSprint, StoryPointInHours = storePointsInHours, Status = status };
+
             var Sprints = await _manageSprintsService.GetAllSprintsAsync(teamId, new DisplayOptions());
-            var activeSprint = Sprints.FirstOrDefault(i => i.IsActive == true);
+            var activeSprint = Sprints.FirstOrDefault(i => i.Status == 1);
```
Comment (alehkahnovich) [inline src/Teams.Web/Controllers/ManageSprintsController.cs:168]: Maybe we need to store Status values as a constants? It is not clear from the code what 1 mean

**Review by alehkahnovich — CHANGES_REQUESTED** [review]

File: src/Teams.Web/Controllers/ManageSprintsController.cs (hunk 1 above)
Comment (sub13) [reply src/Teams.Web/Controllers/ManageSprintsController.cs]: i have fixed this. Please look at the new commit

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Teams.Data.Migrations
+{
+    public partial class AddedStatusToSprint : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropForeignKey(
+                name: "FK_Sprint_Task_TaskId",
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:10]: Please check you migration, why it is here?

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                table: "Sprint");
+
+            migrationBuilder.DropIndex(
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:13]: same

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                name: "IX_Sprint_TaskId",
+                table: "Sprint");
+
+            migrationBuilder.DropColumn(
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:17]: correct

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                name: "IsActive",
+                table: "Sprint");
+
+            migrationBuilder.DropColumn(
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:21]: why?

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                name: "StorePointInHours",
+                table: "Sprint");
+
+            migrationBuilder.DropColumn(
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:25]: Why?

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                name: "TaskId",
+                table: "Sprint");
+
+            migrationBuilder.AddColumn<int>(
+                name: "Status",
+                table: "Sprint",
+                nullable: false,
+                defaultValue: 0);
+
+            migrationBuilder.AddColumn<int>(
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:35]: Why?

File: src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+            migrationBuilder.DropColumn(
+                name: "StorePointInHours",
+                table: "Sprint");
+
+            migrationBuilder.DropColumn(
+                name: "TaskId",
+                table: "Sprint");
+
+            migrationBuilder.AddColumn<int>(
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201202135539_AddedStatusToSprint.cs:29]: Correct

File: src/Teams.Data/Models/PossibleStatuses.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Teams.Data.Models
+{
+    public class PossibleStatuses
+    {
+        public const int createdStatus = 2;
```
Comment (SergeyMokin) [inline src/Teams.Data/Models/PossibleStatuses.cs:9]: maybe better to invert order?
0 is created

File: src/Teams.Data/Migrations/20201203113546_AddedConstVariablesForStatuses.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Teams.Data.Migrations
+{
+    public partial class AddedConstVariablesForStatuses : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
Comment (SergeyMokin) [inline src/Teams.Data/Migrations/20201203113546_AddedConstVariablesForStatuses.cs:9]: why it is here and empty?

**Review by SergeyMokin — CHANGES_REQUESTED** [review]

**Comment by sub13** [comment]
Thank you for your comments. I fixed this problems. Please look at the last commit

**Review by Artem-Halvita — APPROVED** [review]

**Review by SherbaKirill — APPROVED** [review]

File: Tests/Teams.Data.Tests/SprintRepositoryTest.cs [hunk 11]
```diff
@@ -8,7 +8,7 @@
 using Teams.Data.Models;
 using Teams.Data.Repository;
 
-namespace Teams.Data.Tests
+namespace Teams.Business.Tests
```
[the remark this answers is not in the document]
Comment (sub13) [reply Tests/Teams.Data.Tests/SprintRepositoryTest.cs:11]: Yes, you are right. Sorry, i have already changed namespace.

**Review by desenzo — APPROVED** [review]

**Review by Y-floryanovich — APPROVED** [review]

**Review by SergeyMokin — APPROVED** [review]

**Review by aparkhots — APPROVED** [review]

**Review by alehkahnovich — APPROVED** [review]

**Pull request merged**